### DISCUSSION TITLE
Possible issue with <input type="file" capture ...> on Google Pixel running Android 14

### DISCUSSION THREAD

**Discussion by Fittocode** [react-native-webview/react-native-webview#3300] [General]
My team has noticed that our application seems to only experience issues with the input type='file' capture element for pixel users running android 14 (previously all android users but version 13.2.0 solved that https://github.com/react-native-webview/react-native-webview/pull/2954). 

Is anyone aware of any reason why the pixel + android 14 combination in particular might fail to prompt the user to choose one or more files from their device storage?

Here is the html input element being used, which works for all devices except pixels running android 14:

```
 <input
       id="camera"
       type="file"
       name="picture"
       accept="image/*"
       capture={'' as any}
       ref={inputFileRef}
       onChange={(e) => handlePhotoChanges(e)}
/>
```

In handlePhotoChanges is a link that fires but fails to open the camera: `inputFileRef.current.click()`;

Edit: Still trying to root cause in case the issue is unrelated to anything react-native-webview, but it may be that android 14 updated the Storage Access Framework (SAF) to be more stricter about its file access, including via the html input element.

**Comment by diachedelic** [comment]
There was a similar problem in another framework, see the analysis: https://github.com/ionic-team/capacitor/issues/7411.

https://github.com/react-native-webview/react-native-webview/blob/7420ed8e02291a7749d34c6f2c95b1a077319545/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModuleImpl.java#L258-L262

I am betting that `resolveActivity` is returning `null` here on a Google Pixel running Android 14.

**Comment by rlueder** [comment]
(The report mentioned above https://github.com/ionic-team/capacitor/issues/7411 is closed for comments so commenting here in case anyone comes across this)

As of Oc. '24 this doesn't seem to be an issue (see GIF below) and a small repo here: https://github.com/rlueder/android14-troubleshoot

I'm using Capacitor as it's my use case but the test scenario is the same: using input fields to access camera and photo library on a Pixel 6 Pro running Android 14 works as expected.

![android_14_demo](https://github.com/user-attachments/assets/318cebc0-48b4-4abd-aee3-56b0c7b703ed)
